spec/lockfile_spec.rb

```diff
@@ -8,13 +8,13 @@ def self.yaml
       <<-LOCKFILE.strip_heredoc
         PODS:
           - BananaLib (1.0):
-            - monkey (< 1.0.9, ~> 1.0.1)
+            - "monkey (< 1.0.9, ~> 1.0.1)"
```

Comment:
these strings shouldn't need to be quoted?

Reply:
I think this is something `Pod::YAMLHelper` is doing, but the behaviour was previously hidden by the code that removed all quote marks. When we process string values we [escape and quote special characters with `#inspect`](https://github.com/CocoaPods/Core/blob/master/lib/cocoapods-core/yaml_helper.rb#L293-L303) but those quotes are removed afterwards by [the code this PR is changing](https://github.com/CocoaPods/Core/blob/master/lib/cocoapods-core/lockfile.rb#L354-L355). 

What should be the right approach here?